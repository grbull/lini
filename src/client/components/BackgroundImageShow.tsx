import cn from 'classnames';
import React, { ReactElement, ReactNode } from 'react';

import { useIsOnline } from '../hooks/useIsOnline';
import { imageShow } from '../utils/imageShow';

interface Props {
  children: ReactNode;
  imageURL: string | null;
}

export function BackgroundImageShow({
  children,
  imageURL,
}: Props): ReactElement {
  const isOnline = useIsOnline();
  return (
    <div
      className={cn(
        'absolute',
        'bg-scroll',
        'bg-cover',
        'bg-center',
        'bg-scroll',
        'w-full md:max-w-4xl',
        'mx-0 md:mx-auto',
        'h-full',
        'left-0 md:left-auto',
        'top-0',
        'pt-12 md:pt-14',
        { 'pb-14 md:pb-8': isOnline },
        { 'pb-20 md:pb-14': !isOnline }
      )}
      style={{ backgroundImage: `url(${imageShow(imageURL)})` }}
    >
      <div
        className={cn(
          'w-full',
          'h-full',
          'bg-gray-100 dark:bg-gray-700',
          'bg-opacity-30 dark:bg-opacity-80',
          'overflow-y-auto'
        )}
      >
        {children}
      </div>
    </div>
  );
}
