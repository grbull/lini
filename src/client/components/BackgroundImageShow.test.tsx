/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/react';
import React from 'react';

import { BackgroundImageShow } from './BackgroundImageShow';

describe('BackgroundImageShow Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(
      <BackgroundImageShow>Children</BackgroundImageShow>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
