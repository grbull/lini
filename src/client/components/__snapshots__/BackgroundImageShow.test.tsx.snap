// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundImageShow Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="absolute bg-scroll bg-cover bg-center bg-scroll w-full md:max-w-4xl mx-0 md:mx-auto h-full left-0 md:left-auto top-0 pt-12 md:pt-14 pb-14 md:pb-8"
    style="background-image: url(/placeholder_show.png);"
  >
    <div
      class="w-full h-full bg-gray-100 dark:bg-gray-700 bg-opacity-30 dark:bg-opacity-80 overflow-y-auto"
    >
      Children
    </div>
  </div>
</DocumentFragment>
`;
